### PR TITLE
maping through nested rules

### PR DESCRIPTION
**Pull request by makbari**
This is integration of the nested rules. When nested at rules is parsed from grapesjs-parset-postcss, grapes js resolve them. 
an example is also provided in the postcss repo.

### CODE REVIEW

File: src/code_manager/model/CssGenerator.js [hunk 1]
```diff
@@ -107,11 +107,15 @@ export default Backbone.Model.extend({
     const selectorStrNoAdd = rule.selectorsToString({ skipAdd: 1 });
     const selectorsAdd = rule.get('selectorsAdd');
     const singleAtRule = rule.get('singleAtRule');
+    let isGjsRule = false;
```
Comment (artf) [inline src/code_manager/model/CssGenerator.js:110]: what is the point of this variable? Seems like you don't use it...

File: src/parser/model/BrowserParserCss.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const atRules = {
 };
 const atRuleKeys = keys(atRules);
 const singleAtRules = ['5', '6', '11', '15'];
-const singleAtRulesNames = ['font-face', 'page', 'counter-style', 'viewport'];
+const singleAtRulesNames = ['font-face', 'counter-style', 'viewport'];
```
Comment (artf) [inline src/parser/model/BrowserParserCss.js:18]: Why did you remove the `page`?

File: src/code_manager/model/CssGenerator.js [hunk 3]
```diff
@@ -121,8 +125,10 @@ export default Backbone.Model.extend({
       }
     });
 
-    if ((selectorStrNoAdd && found) || selectorsAdd || singleAtRule) {
-      const block = rule.getDeclaration();
+    if ((selectorStrNoAdd && found) || selectorsAdd || singleAtRule || rule.attributes.style) {
```
Comment (artf) [inline src/code_manager/model/CssGenerator.js:128]: `rule.attributes.style` is an object, it's a truthy value, so the condition is pointless

File: src/code_manager/model/CssGenerator.js [hunk 4]
```diff
@@ -121,8 +125,10 @@ export default Backbone.Model.extend({
...
+      const entries = Object.entries(rule.attributes.style);
+      const styleStr = entries.length > 0 ? entries.map(([k, v]) => `${k}:${v}`).join(';') + ';' : '';
+      const block = rule.getDeclaration().length > 0 ? rule.getDeclaration() : styleStr;
```
Comment (artf) [inline src/code_manager/model/CssGenerator.js:131]: You're calling rule.getDeclaration() twice and honestly, I don't understand the point of this change

**Review by artf — CHANGES_REQUESTED** [review]
Can you please explain the point of this PR? What are you trying to solve?

**Comment by makbari** [comment]
Thanks for the review. I will make some correction and more updates. 
There problem I am trying to solve is that, page at rule, can have nested at rule. 
The example is also in https://github.com/artf/grapesjs-parser-postcss/pull/1

**Pull request closed without merging**